components/esm/doorstate.cpp

```diff
@@ -12,12 +14,20 @@ namespace ESM
 
         mDoorState = 0;
         esm.getHNOT (mDoorState, "ANIM");
+        if (mDoorState < 0 || mDoorState > 2)
+            Log(Debug::Warning) << "Skipping an invalid door state (" << mDoorState << ") for door \"" << mRef.mRefID << "\"";
     }
 
     void DoorState::save(ESMWriter &esm, bool inInventory) const
     {
         ObjectState::save(esm, inInventory);
 
+        if (mDoorState < 0 || mDoorState > 2)
+        {
+            Log(Debug::Warning) << "Skipping an invalid door state (" << mDoorState << ") for door \"" << mRef.mRefID << "\"";
```

Comment:
Drop the "an" article in both of the warnings, it shouldn't be necessary (a door only has one state).

Maybe use "dropping" instead of "skipping".

Are warnings in ESM reading/writing code even a thing?

Reply:
> Are warnings in ESM reading/writing code even a thing?

Yes.